src/bindings/python/src/pyopenvino/utils/utils.hpp

```diff
@@ -32,6 +32,28 @@
 
 namespace py = pybind11;
 
+// Conditional GIL management for PEP 703 (free-threaded Python)
+#if defined(Py_GIL_DISABLED) && Py_GIL_DISABLED
+struct gil_scoped_release_if_gil {
+    gil_scoped_release_if_gil() {}
+};
+struct gil_scoped_acquire_if_gil {
+    gil_scoped_acquire_if_gil() {}
+};
+// For free-threaded Python, we don't need call_guard - use empty call_guard
+#define CALL_GUARD_GIL_RELEASE_IF_GIL pybind11::call_guard<>()
```

Comment:
Consider use alias instead macro:
```suggestion
using call_guard_gil_scoped_release = std::conditional_t<PY_GIL_DISABLED , pybind11::call_guard<>, pybind11::call_guard<pybind11::gil_scoped_release>>
```
and use it like normal funciton in code
```cpp
call_guard_gil_scoped_release();
```